documentation/jetty-documentation/src/main/java/org/eclipse/jetty/docs/programming/migration/ServletToHandlerDocs.java

```diff
@@ -0,0 +1,643 @@
+                        {
+                            request.demand(this);
+                            return;
+                        }
+
+                        // If a failure is read, complete with a failure.
+                        if (Content.Chunk.isFailure(chunk))
+                        {
+                            Throwable failure = chunk.getFailure();
+                            completeExceptionally(failure);
+                            return;
+                        }
+
+                        if (chunk instanceof Trailers trailers)
+                        {
+                            // Possibly process the request trailers here.
+                            // Trailers have an empty ByteBuffer and are a last chunk.
+                        }
+
+                        // Process the request content chunk here.
+                        // After the processing, the chunk MUST be released.
+                        chunk.release();
+
+                        // If the last chunk is read, complete normally.
+                        if (chunk.isLast())
+                        {
+                            complete(null);
+                            return;
+                        }
+
+                        // Not the last chunk of content, loop around to read more.
+                    }
+                }
+            };
+
+            // Initiate the read of the request content.
+            reader.start();
+
+            // When the read is complete, complete the Handler callback.
+            callback.completeWith(reader);
+
+            return true;
+        }
+        // end::requestContent-source[]
+    }
+
+    @SuppressWarnings("InnerClassMayBeStatic")
+    // tag::response[]
+    public class ResponseAPIs extends Handler.Abstract
+    {
+        @Override
+        public boolean handle(Request request, Response response, Callback callback) throws Exception
+        {
+            // Sets/Gets the response HTTP status.
+            // Replaces:
+            //   - servletResponse.setStatus(code);
+            //   - servletResponse.getStatus();
+            response.setStatus(HttpStatus.OK_200);
+            int status = response.getStatus();
+
+            // Gets the response HTTP headers.
+            // Replaces:
+            //   - servletResponse.setHeader(name, value);
+            //   - servletResponse.addHeader(name, value);
+            //   - servletResponse.setDateHeader(name, date);
+            //   - servletResponse.addDateHeader(name, date);
+            //   - servletResponse.setIntHeader(name, value);
+            //   - servletResponse.addIntHeader(name, value);
+            //   - servletResponse.getHeaderNames()
+            //   - servletResponse.getHeader(name)
+            //   - servletResponse.getHeaders(name)
+            //   - servletResponse.containsHeader(name)
+            HttpFields.Mutable responseHeaders = response.getHeaders();
```

Comment:
maybe a few examples to show that we can operate on HttpFields - so we can set precomputed ones, do computes etc.